src/inspector.rs

```diff
@@ -271,6 +271,9 @@ where
 pub trait ChannelImpl: AsChannel {
   fn base(&self) -> &ChannelBase;
   fn base_mut(&mut self) -> &mut ChannelBase;
+  fn base_ptr(this: *const Self) -> *const ChannelBase
+  where
+    Self: Sized;
```

Comment:
Question: should this function be an `unsafe fn`? I think this function *has* to be implemented with `addr_of` for correctness, which has safety requirements (the `expr` must follow all the general rules, for example no null pointer deref is allowed), so calling `T::base_ptr(std::ptr::null_mut())` would be UB in safe code.

Also, maybe a comment could be added to the function so implementors know what to do or what this is for, like mentioning that this is used for calculating the offset to the base field, and that care must be taken not to create any references in the process of creating the pointer because the `*const Self` pointer is not valid (thus resulting in instant UB). Maybe a link to an implementation could be useful?

Reply:
I believe that this should be a unsafe fn, as the implementation is unsafe. I left a comment along in the implementation.